### PR TITLE
MOEN-23409: Segment-MoEngage PluginBase

### PR DESCRIPTION
**Pull request by RakshithaAcharya**

### CODE REVIEW

File: MoEngageSegmentPluginBase/MoEngageSegmentPluginBase/MoEngageSegmentPluginParser.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+//  MoEngageSegmentPluginParser.swift
+//  MoEngageSegmentPluginBase
+//
+//  Created by Rakshitha on 09/05/23.
+//
+
+import Foundation
+import MoEngageSDK
+
+class MoEngageSegmentPluginParser {
```
Comment (umangmoe) [inline MoEngageSegmentPluginBase/MoEngageSegmentPluginBase/MoEngageSegmentPluginParser.swift:11]: Do we really need to keep the methods static? I would always be in memory.

**Review by umangmoe — CHANGES_REQUESTED** [review]
For the test cases, have the test data and expected results in a separate file.

File: MoEngageSegmentPluginBase/MoEngageSegmentPluginBase/MoEngageSegmentPluginParser.swift (hunk 1 above)
Comment (RakshithaAcharya) [reply MoEngageSegmentPluginBase/MoEngageSegmentPluginBase/MoEngageSegmentPluginParser.swift:11]: @umangmoe  done

**Comment by RakshithaAcharya** [comment]
> For the test cases, have the test data and expected results in a separate file.

 Since its small data  we can keep it in same file instead of different

**Pull request merged**